### PR TITLE
put the pipelineID into the request body when the action cancel pipeline

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
put the pipelineID into the request body when the action cancel pipeline

erda-issue: [erda-issue](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/all?id=72603&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDU2MCJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=BUG)

link pr: https://github.com/erda-project/erda-actions/pull/121#issue-702939431

### CODE REVIEW

File: modules/dop/endpoints/pipeline.go [hunk 1]
```diff
@@ -394,8 +409,17 @@ func (e *Endpoints) pipelineCancel(ctx context.Context, r *http.Request, vars ma
...
 	}
 
-	// 根据 pipelineID 获取 pipeline 信息
-	p, err := e.bdl.GetPipeline(pipelineID)
+	// action will cancel pipelineID,  pipelineID not the id that needs to be canceled
+	var cancelRequest apistructs.PipelineCancelRequest
+	if err := json.NewDecoder(r.Body).Decode(&cancelRequest); err != nil {
+		logrus.Errorf("error to decode runRequest")
+	}
+	if r.Header.Get(httputil.InternalActionHeader) == "" {
```
Comment (sfwn) [inline modules/dop/endpoints/pipeline.go:417]: Do not depend on internal action header. Other invokers send request body with `CancelRequest` also should cancel pipeline in body.

Comment (sfwn) [reply modules/dop/endpoints/pipeline.go:417]: Action why not directly invoke with pipelineID in URL path ?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/endpoints/pipeline.go (hunk 1 above)
Comment (kakj-go) [reply modules/dop/endpoints/pipeline.go:417]: image

the pic is erda action request token, the token will check with pipelineID, if send pipelineID was not this id , request will responce 403 error, i add pipelineID in body when action client request

Comment (kakj-go) [reply modules/dop/endpoints/pipeline.go:417]: For this reason, the request needs to identify whether it is from the action request. If it is not from the action request, it should follow the original logic

Comment (sfwn) [reply modules/dop/endpoints/pipeline.go:417]: You are right! you should comment it in the code.

File: modules/dop/endpoints/pipeline.go [hunk 2]
```diff
@@ -131,6 +131,21 @@ func (e *Endpoints) pipelineDetail(ctx context.Context, r *http.Request, vars ma
 		return apierrors.ErrGetPipeline.InvalidParameter(err).ToResp(), nil
 	}
 
+	identityInfo, err := user.GetIdentityInfo(r)
+	if err != nil {
+		return apierrors.ErrGetUser.InvalidParameter(err).ToResp(), nil
+	}
+
+	// 根据 pipelineID 获取 pipeline 信息
```
Comment (sfwn) [inline modules/dop/endpoints/pipeline.go:139]: use English please.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**